### PR TITLE
Remove Data telemetry option from convert2rhel tests

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
Data telemetry input/option is removed from convert2rhel job templates in https://github.com/theforeman/foreman_ansible/pull/710

### Solution
Remove Data telemetry input/option from convert2rhel tests

Also, Adding check to vefify convert2rhel conversion facts to cover SAT-24654

### CODE REVIEW

**Review by shweta83 — APPROVED** [review]

File: tests/foreman/api/test_convert2rhel.py [hunk 1]
```diff
@@ -299,7 +296,6 @@ def test_convert2rhel_oracle_with_pre_conversion_template_check(
             'inputs': {
                 'Activation Key': activation_key_rhel.id,
                 'Restart': 'yes',
```
Comment (shweta83) [inline tests/foreman/api/test_convert2rhel.py:298]: This change is part of an ongoing feature for Convert2Rhel improvement so I would suggest to attach jira link here as well with `Verifies` docstring.

File: tests/foreman/api/test_convert2rhel.py [hunk 2]
```diff
@@ -355,9 +351,6 @@ def test_convert2rhel_centos_with_pre_conversion_template_check(
         synchronous=False,
         data={
             'job_template_id': template_id,
-            'inputs': {
```
Comment (shweta83) [inline tests/foreman/api/test_convert2rhel.py:358]: This change is part of an ongoing feature for Convert2Rhel improvement so I would suggest to attach jira link here as well with Verifies docstring.

**Review by shweta83 — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_convert2rhel.py (hunk 2 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_convert2rhel.py:358]: its just a removal of input, not addition, so why to keep Verifies inside docstrings, it doesn't make sense, right?

Comment (shweta83) [reply tests/foreman/api/test_convert2rhel.py:358]: This is functionality change in convert2rhel so I am in favor of keeping jira in docstring.

Comment (Gauravtalreja1) [reply tests/foreman/api/test_convert2rhel.py:358]: yes, I understand, but we're not adding new change, its just a removal of existing option which was non-default

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7573
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_convert2rhel.py --external-logging
Test Result : ================= 4 failed, 906 warnings in 6264.96s (1:44:24) =================
```

**Comment by shweta83** [comment]
@Gauravtalreja1 Can you please check the failures on this PR?

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py -k "test_convert2rhel_oracle_with_pre_conversion_template_check[oracle7] or test_convert2rhel_centos_with_pre_conversion_template_check[centos7]"

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7583
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_convert2rhel.py --external-logging
Test Result : ============ 3 failed, 669 warnings, 1 error in 5084.26s (1:24:44) =============
```

File: tests/foreman/api/test_convert2rhel.py [hunk 3]
```diff
@@ -405,3 +405,7 @@ def test_convert2rhel_centos_with_pre_conversion_template_check(
         or host_content['operatingsystem_name'].startswith(f'RedHat {version}')
         or host_content['operatingsystem_name'].startswith(f'RHEL {version}')
     )
+
+    # Vefify convert2rhel facts are generated, and verify fact conversions.success is true
```
Comment (shweta83) [inline tests/foreman/api/test_convert2rhel.py:409]:
```suggestion
    # Verify convert2rhel facts are generated, and verify fact conversions.success is true
```

File: tests/foreman/api/test_convert2rhel.py [hunk 4]
```diff
@@ -321,6 +319,10 @@ def test_convert2rhel_oracle_with_pre_conversion_template_check(
         or host_content['operatingsystem_name'].startswith(f'RHEL {version}')
     )
 
+    # Vefify convert2rhel facts are generated, and verify fact conversions.success is true
```
Comment (shweta83) [inline tests/foreman/api/test_convert2rhel.py:322]:
```suggestion
    # Verify convert2rhel facts are generated, and verify fact conversions.success is true
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_convert2rhel.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7639
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_convert2rhel.py --external-logging
Test Result : ======= 1 failed, 1 passed, 616 warnings, 2 errors in 4805.17s (1:20:05) =======
```

**Comment by Gauravtalreja1** [comment]
PRT passed for centos8 conversion, while other conversions fails due to intermittent issues, so I'd request to review/merge

**Review by shweta83 — APPROVED** [review]
ACK

**Comment by shweta83** [comment]
The scope of this PR has been completed and passing and the failures are unrelated to this change so it should be good to merge this PR.

**Review by shubhamsg199 — APPROVED** [review]
Ack

**Pull request merged**